language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/AddToConfigTomlCodeAction.java

```diff
@@ -0,0 +1,288 @@
+import org.ballerinalang.langserver.commons.codeaction.spi.RangeBasedCodeActionProvider;
+import org.ballerinalang.langserver.commons.codeaction.spi.RangeBasedPositionDetails;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.Command;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Code Action to undefined configurables into Config.toml file.
+ *
+ * @since 2201.10.0
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class AddToConfigTomlCodeAction implements RangeBasedCodeActionProvider {
+
+    private static final String CONFIG_TOML = "Config.toml";
+
+    @Override
+    public List<SyntaxKind> getSyntaxKinds() {
+        return List.of(SyntaxKind.MODULE_VAR_DECL);
+    }
+
+    @Override
+    public boolean validate(CodeActionContext context, RangeBasedPositionDetails positionDetails) {
+        return context.currentModule().isPresent()
+                && hasConfigurableQualifier((ModuleVariableDeclarationNode) positionDetails.matchedCodeActionNode());
+    }
+
+    @Override
+    public List<CodeAction> getCodeActions(CodeActionContext context, RangeBasedPositionDetails posDetails) {
+        Module module = context.currentModule().get();
+        Project project = module.project();
+        if (project.kind() == ProjectKind.SINGLE_FILE_PROJECT) {
+            return Collections.emptyList();
+        }
+        String orgName = project.currentPackage().packageOrg().value();
+        SemanticModel semanticModel = context.currentSemanticModel().get();
+        Types types = semanticModel.types();
+        TypeBuilder builder = types.builder();
+        TypeSymbol basicType = builder.UNION_TYPE.withMemberTypes(types.INT, types.STRING, types.BOOLEAN,
+                types.BYTE, types.FLOAT, types.DECIMAL, types.XML).build();
+        TypeSymbol anyDataOrJsonType = builder.UNION_TYPE.withMemberTypes(types.ANYDATA, types.JSON).build();
+
+        Optional<Symbol> confSymbol = semanticModel.symbol(posDetails.matchedCodeActionNode());
+        if (confSymbol.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        ConfigurableFinder.ModuleConfigDetails moduleConfigDetails = ConfigurableFinder.getModuleConfigDetails(context);
+        Map<String, ConfigurableFinder.ConfigVariable> configVarMap = moduleConfigDetails.configVariables();
+        VariableSymbol variableSymbol = (VariableSymbol) confSymbol.get();
+        ConfigurableFinder.ConfigVariable selectedConf = configVarMap.get(variableSymbol.getName().get());
+
+        Path configTomlPath = context.workspace().projectRoot(context.filePath()).resolve(CONFIG_TOML);
```

Comment:
We can configure the config toml path with the env var `BAL_CONFIG_FILES` [1]. Shouldn't we check its availability before assuming the path to be `root/Config.toml`?
[1] https://ballerina.io/learn/provide-values-to-configurable-variables/#provide-via-configuration-files

Reply:
There are few things needs to consider in this case. 

1. The Ballerina VS Code plugin only controls the project workspace. If the Config.toml file that the user intends to use is outside the project workspace, the VS Code plugin cannot make changes to it.

2. This feature is primarily useful during the development phase of a program. Users typically do not use this environment variable to specify configs during development. In test or production environments, they probably provide the config path using environment variables.

3. The main purpose of this code action is to enhance the developer experience (DX) during development. It allows users to create a `Config.toml` file using the code action and add missing elements to the Config.toml file. We also provide default values for each config variable. Therefore, if users want to run the program correctly during runtime, they must update the values in the `Config.toml` file.

For the above reasons we have thought, it is fine not to consider whether a `Config.toml` file exists in a different location.

Reply:
Yeah in that case, it makes sense to assume Config.toml to be there in the root. However, IMO, it's better to check if BAL_CONFIG_FILES is set and print a warning at least.

Reply:
One way we can warn user using the OUTPUT bar of the vscode. But that place is not very user friendly. Can you create a issue by specifying this?